### PR TITLE
Review API

### PR DESCRIPTION
**Pull request by nopparujp**
- add review
- add average rating

### CODE REVIEW

File: models/auth_model.go [hunk 1]
```diff
@@ -14,11 +14,11 @@ import (
 var userCollection *mongo.Collection = configs.GetCollection(configs.DB, "users")
 
 type TrainerInfo struct {
-	Specialty      []string `bson:"specialty"      json:"specialty"`
-	Fee            int      `bson:"fee,omitempty"            json:"fee"`
-	CertificateURL string   `bson:"certificateUrl,omitempty" json:"certificateUrl"`
-	Rating         float64  `bson:"rating,omitempty"         json:"rating"`
-	TraineeCount   int64    `bson:"traineeCount,omitempty"   json:"traineeCount"`
+	Specialty      []string `bson:"specialty"`
+	Fee            float64  `bson:"fee"`
```
Comment (noppakorn) [inline models/auth_model.go]: Money should be always be in Decimal. https://pkg.go.dev/github.com/shopspring/decimal

File: models/trainer_model.go [hunk 2]
```diff
@@ -50,6 +51,78 @@ type FilteredTrainerInfo struct {
 	AvatarUrl   string      `json:"avatarUrl"`
 	TrainerInfo TrainerInfo `json:"trainerInfo"`
 }
+type Review struct {
+	Username        string    `bson:"username"`
+	Rating          float64   `bson:"rating"`
+	Comment         string    `bson:"comment"`
+	ReviewCreatedAt time.Time `bson:"reviewCreatedAt"`
```
Comment (noppakorn) [inline models/trainer_model.go:58]: Just createdAt should be fine?

File: routes/protected_routes.go [hunk 3]
```diff
@@ -25,5 +25,9 @@ func ProtectedRoute(router *gin.Engine) {
 	// Get Others Trainer information
 	protected.POST("/trainer", controllers.GetTrainerProfile())
 	protected.POST("/filter-trainer", controllers.FilterTrainer())
+	// Get review
+	protected.POST("/get-reviews", controllers.GetReviews())
```
Comment (noppakorn) [inline routes/protected_routes.go]: Why long name? "reviews" should be fine no? This is to not confuse with "GET" request

File: models/trainer_model.go [hunk 4]
```diff
@@ -263,3 +255,69 @@ func GetDistance(lat1 float64, lng1 float64, lat2 float64, lng2 float64) float64
...
+	defer cursor.Close(context.Background())
+	// var reviews []Review
+	reviews := make([]Review, 0)
+	// Iterate through the results
+	for cursor.Next(context.Background()) {
+		// var result bson.M
+		err := cursor.Decode(&result)
+		if err != nil {
+			fmt.Println(err)
+		}
```
Comment (noppakorn) [inline models/trainer_model.go]: Handle this error.

**Review by noppakorn — CHANGES_REQUESTED** [review]

File: models/trainer_model.go (hunk 2 above)
Comment (nopparujp) [reply models/trainer_model.go:58]: Fixed in booking branch

**Review by noppakorn — APPROVED** [review]

**Pull request merged**